frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -119,14 +118,18 @@ common_settings.screen = {
     },
 }
 if Screen.isColorScreen() then
-    table.insert(common_settings.screen.sub_item_table, 4, require("ui/elements/screen_color_menu_table"))
-    common_settings.screen.sub_item_table[4].separator = true
-else
-    common_settings.screen.sub_item_table[3].separator = true
+    table.insert(common_settings.screen.sub_item_table, 2, require("ui/elements/screen_color_menu_table"))
+    basicSettings = basicSettings + 1
 end
 if Device:isAndroid() then
-    table.insert(common_settings.screen.sub_item_table, require("ui/elements/screen_fullscreen_menu_table"))
+    table.insert(common_settings.screen.sub_item_table, 2, require("ui/elements/screen_fullscreen_menu_table"))
+    basicSettings = basicSettings + 1
+elseif Device:isKobo() or Device:isKindle() or Device:isPocketBook() then
```

Comment:
It's better to add something to Device or Screen about this, akin to the `isColorScreen()` you can see just a few lines above. That way it's clearer what's supported by which device, plus it's easier to turn things on and off for testing because you won't have to go looking all over the code. You can just flip the switch.

Case in point: we probably do want to show this menu on the emulator, at the very least in full-on flash emulation mode, and since I found [this Onyx API](https://github.com/onyx-intl/OnyxAndroidSample/blob/8b166ac604b48fb67c124957cd5432240c7050fd/doc/EPD-Screen-Update.md) and we've also seen [this Tolino backwards engineering](https://github.com/koreader/koreader/issues/3517#issuecomment-423714511) there is Android potential as well.

Those things should be abstracted at a higher level than this menu generation code.

Besides that I think this looks good. Pinging @poire-z for a second opinion.

Reply:
I fully agree with your comment and suggestions.
(There are indeed Android devices that are eInk, and we probably want this menu item on the emulator - or a quick way to re-enable it - for testing full refreshes counter and such. And not only with full-on flash, I often just look at the log/console output to see what kind of refresh has been done.)

Reply:
Thanks for the feedback @Frenzie @poire-z 

> It's better to add something to Device or Screen about this, akin to the isColorScreen() you can see just a few lines above. That way it's clearer what's supported by which device, plus it's easier to turn things on and off for testing because you won't have to go looking all over the code. You can just flip the switch.

Agree. I see https://github.com/koreader/koreader/blob/master/frontend/device/generic/device.lua#L72-L73. We can create a new  isEinkScreen() which returns true on emulator. I have no idea how to differenciate between android with and without eink screens. Any hint on that?. After a quick search I found https://stackoverflow.com/questions/15221787/android-how-to-detect-an-e-ink-screen

> Case in point: we probably do want to show this menu on the emulator, at the very least in full-on flash emulation mode, and since I found this Onyx API and we've also seen this Tolino backwards engineering there is Android potential as well.

[EpdDeviceManager](https://github.com/onyx-intl/OnyxAndroidSample/blob/8b166ac604b48fb67c124957cd5432240c7050fd/doc/EpdDeviceManager.md) seems compatible with RK3026 and IMX6 devices.  There is a [demo](https://github.com/onyx-intl/OnyxAndroidSample/blob/8b166ac604b48fb67c124957cd5432240c7050fd/app/sample/src/main/java/com/onyx/android/sample/EpdDemoActivity.java) which shows how to use the wrapper.

Reply:
> I have no idea how to differenciate between android with and without eink screens. Any hint on that?

As far as this PR's concerned, there's no such thing as Android with E Ink. It was merely an example of something that'd later require unnecessary additional maintenance effort to implement if the PR were merged in its current state. :-)